### PR TITLE
[TK-08547] Tx2 Error Handling

### PR DESCRIPTION
**Pull request by neonphog**
should be merged into https://github.com/holochain/holochain/pull/731 before that is merged into develop

- additional tracing
- additional error handling
- proxy routing errors reflected back to original requestor
- some additional metrics
- proxy connection management fixes

### CODE REVIEW

File: crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs [hunk 1]
```diff
@@ -808,17 +808,17 @@ impl Space {
...
             //race_timeout_ms,
             payload,
             ..
         } = input;
-        let remote_agent_count = remote_agent_count.unwrap();
-        let timeout_ms = timeout_ms.unwrap();
-        let stage_1_timeout_ms = timeout_ms / 2;
+        //let remote_agent_count = remote_agent_count.unwrap();
+        //let timeout_ms = timeout_ms.unwrap();
+        //let stage_1_timeout_ms = 2000;
```
Comment (freesig) [inline crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs:821]: Was this on purpose?

File: crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs [hunk 2]
```diff
@@ -876,8 +878,10 @@ impl Space {
                     }
                 })
                 .collect();
+            //tracing::warn!("local_all count: {}", out.len());
 
-            out.append(&mut remote_fut.await);
+            //out.append(&mut remote_fut.await);
+            //tracing::warn!("plus remote count: {}", out.len());
```
Comment (freesig) [inline crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs:884]: Maybe delete if not needed?

File: crates/kitsune_p2p/proxy/src/tx2.rs [hunk 3]
```diff
@@ -360,29 +444,35 @@ async fn incoming_evt_handle(
...
-                    let dest_digest = CertDigest(Arc::new(data[DEST_START..DEST_END].to_vec()));
-                    //println!("src: {:?}", src_digest);
-                    //println!("dst: {:?}", dest_digest);
-                    //println!("loc: {:?}", hnd.local_digest);
-                    if dest_digest == hnd.local_digest {
+                    let src_cert = data[SRC_START..SRC_END].to_vec().into();
+                    let dest_cert = data[DEST_START..DEST_END].to_vec().into();
+                    //println!("src: {:?}", src_cert);
+                    //println!("dst: {:?}", dest_cert);
+                    //println!("loc: {:?}", hnd.local_cert);
```
Comment (freesig) [inline crates/kitsune_p2p/proxy/src/tx2.rs:464]: Maybe remove these as well?

File: crates/kitsune_p2p/types/src/config.rs [hunk 4]
```diff
@@ -83,17 +83,17 @@ pub mod tuning_params_struct {
...
 
-        /// Default timeout for remote notify. [Default: 1000ms]
-        default_notify_timeout_ms: u32 = 1000,
+        /// Default timeout for remote notify. [Default: 30s]
+        default_notify_timeout_ms: u32 = 1000 * 30,
 
-        /// Default timeout for rpc single. [Default: 2000]
-        default_rpc_single_timeout_ms: u32 = 2000,
+        /// Default timeout for rpc single. [Default: 30s]
+        default_rpc_single_timeout_ms: u32 = 1000 * 30,
```
Comment (freesig) [inline crates/kitsune_p2p/types/src/config.rs:90]: Why such large jumps in timeout defaults?

**Review by freesig — APPROVED** [review]
Looks good, just a few commented out bits of code that might be able to be removed.

File: crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs (hunk 1 above)
Comment (neonphog) [reply crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs]: there was some experimentation, i reverted it : )

File: crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs (hunk 2 above)
Comment (neonphog) [reply crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs]: :+1:

File: crates/kitsune_p2p/proxy/src/tx2.rs (hunk 3 above)
Comment (neonphog) [reply crates/kitsune_p2p/proxy/src/tx2.rs]: :+1:

File: crates/kitsune_p2p/types/src/config.rs (hunk 4 above)
Comment (neonphog) [reply crates/kitsune_p2p/types/src/config.rs:90]: 2 seconds is not reasonable for network requests - perhaps 30s is too large, but I feel like it's a good plate to start.

**Pull request merged**